test/CMakeLists.txt

```diff
@@ -161,7 +161,8 @@ macro(add_rev_test test_name test_dir timeout labels)
     # Custom target for running the specified script
     add_custom_target(run_${test_name_lower}_script
       COMMAND ${CMAKE_COMMAND} -E echo "Executing script ${optional_script}"
-      COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR}/${test_name_lower} && bash ${optional_script}
+      COMMAND bash ${optional_script}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${test_dir_lower}
```

Comment:
`WORKING_DIRECTORY` should be set instead of using `cd` to change the directory, and the directory should be `${test_dir_lower}` instead of `${test_name_lower}`.